cypress/component/highlight_text.spec.js

```diff
@@ -0,0 +1,174 @@
+    cy.get('[data-cy="highlight-0"]').should('have.css', 'color', 'rgb(255, 255, 255)');
+
+    cy.get('[data-cy="highlight-1"]').contains(/^ $/);
+    // default, unchanged CSS props
+    cy.get('[data-cy="highlight-1"]').should('have.css', 'background-color', 'rgba(0, 0, 0, 0)');
+    cy.get('[data-cy="highlight-1"]').should('have.css', 'color', 'rgb(0, 0, 0)');
+
+    cy.get('[data-cy="highlight-2"]').contains(/^text$/);
+    // blue highlighted CSS props
+    cy.get('[data-cy="highlight-2"]').should('have.css', 'background-color', 'rgb(0, 255, 0)');
+    cy.get('[data-cy="highlight-2"]').should('have.css', 'color', 'rgb(255, 0, 0)');
+
+
+    cy.get('[data-cy="highlight-3"]').contains(/^. $/);
+    // default, unchanged CSS props
+    cy.get('[data-cy="highlight-3"]').should('have.css', 'background-color', 'rgba(0, 0, 0, 0)');
+    cy.get('[data-cy="highlight-3"]').should('have.css', 'color', 'rgb(0, 0, 0)');
+
+
+    cy.get('[data-cy="highlight-4"]').contains(/^Test$/);
+    // black highlighted CSS props
+    cy.get('[data-cy="highlight-4"]').should('have.css', 'background-color', 'rgb(0, 0, 0)');
+    cy.get('[data-cy="highlight-4"]').should('have.css', 'color', 'rgb(255, 255, 255)');
+  });
+
+
+  it('Does not highlight word if it isn\'t surrounded by whitespace or punctuation.', function() {
+    const text = (
+      'Example; asdfexample exampleasdf Words. multiple words. ahhhhmultiple words'
+    );
+
+    mount(
+      <HighlightText
+        highlight={[
+          // black bg, white text
+          {
+            bgColor: 'rgb(0, 0, 0)',
+            textColor: 'rgb(255, 255, 255)',
+            words: ['example', 'multiple words'],
+          },
+        ]}
+        text={text}
+      />
+    );
+
+    cy.get('[data-cy="highlight-0"]').contains(/^Example$/);
+    // black highlighted CSS props
+    cy.get('[data-cy="highlight-0"]').should('have.css', 'background-color', 'rgb(0, 0, 0)');
+    cy.get('[data-cy="highlight-0"]').should('have.css', 'color', 'rgb(255, 255, 255)');
+
+
+    cy.get('[data-cy="highlight-1"]').contains(/^; asdfexample exampleasdf Words. $/);
+    // default, unchanged CSS props
+    cy.get('[data-cy="highlight-1"]').should('have.css', 'background-color', 'rgba(0, 0, 0, 0)');
+    cy.get('[data-cy="highlight-1"]').should('have.css', 'color', 'rgb(0, 0, 0)');
+
+    cy.get('[data-cy="highlight-2"]').contains(/^multiple words$/);
+    // black highlighted CSS props
+    cy.get('[data-cy="highlight-2"]').should('have.css', 'background-color', 'rgb(0, 0, 0)');
+    cy.get('[data-cy="highlight-2"]').should('have.css', 'color', 'rgb(255, 255, 255)');
+
+
+    cy.get('[data-cy="highlight-3"]').contains(/^. ahhhhmultiple words$/);
+    // default, unchanged CSS props
+    cy.get('[data-cy="highlight-3"]').should('have.css', 'background-color', 'rgba(0, 0, 0, 0)');
+    cy.get('[data-cy="highlight-3"]').should('have.css', 'color', 'rgb(0, 0, 0)');
+
+  });
+});
```

Comment:
Should words that are substrings of other words match?

For example if:
`words: ['doctor','cancer']`

and

`text={'The doctors found many types of cancers growing'}`

Would 'doctor' be highlighted in 'doctors' and would 'cancer' be highlighted in 'cancers'?